drivers/mrf24j40/mrf24j40_getset.c

```diff
@@ -0,0 +1,499 @@
+                tmp |= MRF24J40_TXMCR_NOCSMA;
+                /* MACMINBE<1:0>: The minimum value of the backoff exponent
+                 * in the CSMA-CA algorithm. Note that if this value is set
+                 * to �0�, collision avoidance is disabled. */
+                mrf24j40_reg_write_short(dev, MRF24J40_REG_TXMCR, tmp);
+
+                break;
+            case MRF24J40_OPT_PROMISCUOUS:
+                DEBUG("[mrf24j40] opt: disabling PROMISCUOUS mode\n");
+                /* disable promiscuous mode */
+                tmp = mrf24j40_reg_read_short(dev, MRF24J40_REG_RXMCR);
+                tmp &= ~MRF24J40_RXMCR_PROMI;
+                tmp &= ~MRF24J40_RXMCR_ERRPKT;
+                /* re-enable AUTOACK only if the option is set */
+                if (dev->netdev.flags & NETDEV2_IEEE802154_ACK_REQ) {
+                    tmp &= ~(MRF24J40_RXMCR_NOACKRSP);
+                    mrf24j40_reg_write_short(dev, MRF24J40_REG_RXMCR, tmp);
+                }
+                break;
+            case NETDEV2_IEEE802154_ACK_REQ:
+                DEBUG("[mrf24j40] opt: disabling auto ACKs\n");
+                tmp = mrf24j40_reg_read_short(dev, MRF24J40_REG_RXMCR);
+                tmp |= MRF24J40_RXMCR_NOACKRSP;
+                mrf24j40_reg_write_short(dev, MRF24J40_REG_RXMCR, tmp);
+                break;
+            default:
+                /* do nothing */
+                break;
+        }
+    }
+}
+
+
+void mrf24j40_set_state(mrf24j40_t *dev, uint8_t state)
+{
+    uint8_t old_state;
+
+    old_state = dev->state;
+
+
+    if (state == old_state) {
+        return;
+    }
+    /* check if asked to wake up from sleep mode */
+    if (old_state == MRF24J40_PSEUDO_STATE_SLEEP) {
+        mrf24j40_assert_awake(dev);
+    }
+    if (state == MRF24J40_PSEUDO_STATE_SLEEP) {
+        mrf24j40_sleep(dev);
+    }
+    if (state == MRF24J40_PSEUDO_STATE_IDLE) {
+        dev->state = state;
+    }
+    dev->idle_state = state;
+}
+
+void mrf24j40_sleep(mrf24j40_t *dev)
+{
+    DEBUG("[mrf24j40] Putting into sleep mode\n");
+    /* Datasheet chapter 3.15.2 IMMEDIATE SLEEP AND WAKE-UP MODE */
+    /* First force a Power Management Reset */
+    mrf24j40_reg_write_short(dev, MRF24J40_REG_SOFTRST, MRF24J40_SOFTRST_RSTPWR);
+    /* Go to SLEEP mode */
+    mrf24j40_reg_write_short(dev, MRF24J40_REG_SLPACK, MRF24J40_SLPACK_SLPACK);
+    dev->state = MRF24J40_PSEUDO_STATE_SLEEP;
+}
+
+void mrf24j40_assert_sleep(mrf24j40_t *dev)
+{
+    if (dev->idle_state == MRF24J40_PSEUDO_STATE_SLEEP) {
+        mrf24j40_sleep(dev);
+    }
+}
+
+void mrf24j40_assert_awake(mrf24j40_t *dev)
+{
+    if (dev->state == MRF24J40_PSEUDO_STATE_SLEEP) {
+        DEBUG("[mrf24j40] Waking up from sleep mode\n");
+        /* Wake mrf up */
+        mrf24j40_reg_write_short(dev, MRF24J40_REG_WAKECON, MRF24J40_WAKECON_IMMWAKE | MRF24J40_WAKECON_REGWAKE);
+        /* undocumented delay, needed for stable wakeup */
+        xtimer_usleep(50);
```

Comment:
Hmmm. Who added this? And how was it observed? Did anyone compare to the linux driver?